search_user_blackbox_test.go

```diff
@@ -0,0 +1,204 @@
+package main_test
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"testing"
+
+	"strings"
+
+	. "github.com/almighty/almighty-core"
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/app/test"
+	"github.com/almighty/almighty-core/application"
+	"github.com/almighty/almighty-core/gormapplication"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/goadesign/goa"
+	"golang.org/x/net/context"
+)
+
+type args struct {
+	pageOffset *string
+	pageLimit  *int
+	q          string
+}
+
+type expect func(*testing.T, okScenario, *app.SearchResponseUsers)
+type expects []expect
+
+type okScenario struct {
+	name    string
+	args    args
+	expects expects
+}
+
+func TestUsersSearchOK(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	idents := createTestData()
+	defer cleanTestData(idents)
+
+	tests := []okScenario{
+		{"With uppercase fullname query", args{offset("0"), limit(10), "TEST_AB"}, expects{totalCount(1)}},
+		{"With uppercase fullname query", args{offset("0"), limit(10), "TEST_AB"}, expects{totalCount(1)}},
+		{"With uppercase email query", args{offset("0"), limit(10), "EMAIL_TEST_AB"}, expects{totalCount(1)}},
+		{"With lowercase email query", args{offset("0"), limit(10), "email_test_ab"}, expects{totalCount(1)}},
+		{"with special chars", args{offset("0"), limit(10), "&:\n!#%?*"}, expects{totalCount(0)}},
+		{"with * to list all", args{offset("0"), limit(10), "*"}, expects{totalCountAtLeast(len(idents))}},
+		{"with multi page", args{offset("0"), limit(10), "TEST"}, expects{hasLinks("Next")}},
+		{"with last page", args{offset(strconv.Itoa(len(idents) - 1)), limit(10), "TEST"}, expects{hasNoLinks("Next"), hasLinks("Prev")}},
+		{"with different values", args{offset("0"), limit(10), "TEST"}, expects{differentValues()}},
+	}
+
+	service := goa.New("TestUserSearch-Service")
+	controller := NewSearchController(service, gormapplication.NewGormDB(DB))
+
+	for _, tt := range tests {
```

Comment:
I stil don't like this pattern, as it gives me no line number for the failing tests. Why not do it like:
```
runTest({"with last page", args{offset(strconv.Itoa(len(idents) - 1)), limit(10), "TEST"}, expects{hasNoLinks("Next"), hasLinks("Prev")}})
runTest({"with different values", args{offset("0"), limit(10), "TEST"}, expects{differentValues()}})
```
It makes no real difference in the code and gives a line number for the failure.